### DISCUSSION TITLE
Skip.Tools Install Issue : JAVA_HOME is incorrectly set

### DISCUSSION THREAD

**Discussion by tom-eager** [skiptools/skip#200] [Q&A] [answered]
I installed HomeBrew on my Mac with a M1 processor and then installed skip.tools.  When I run skip checkup I get this result: 
![Skip tools failure](https://github.com/user-attachments/assets/dacaa5c8-739e-4026-b2e0-89c877fabde9)

Both xCode and Android Studio work with single platform projects.

I attempted to create an empty multi-platform project and get this result:

```
ERROR: JAVA_HOME is set to an invalid directory: /opt/homebrew/opt/java

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

I have no idea where JAVA_Home is set to "/opt/homebrew/opt/java"  

Suggestions?

**Comment by marcprux** [comment]
If JAVA_HOME is unset in your environment (typically from the ~/.zshrc file), then the `skip` command will automatically try to use the `/opt/homebrew/opt/java` directory, which is where Homebrew typically installs the default java.

It is surprising that this is empty, since `openjdk` is a dependency of `skip`. If you manually install it with `brew install openjdk`, does the `/opt/homebrew/opt/java` folder exist? If you just run `echo $JAVA_HOME` and `java -version` from the command line, what do you see?

**Comment by tom-eager** [comment]
I manually reInstalled openjdk ..
 It was poured into :     /usr/local/Cellar/openjdk/22.0.2
My HomeBrew folder is :  /usr/local/Homebrew

/opt/homebrew/opt/java does not exist. 
echo $JAVA_HOME is now empty

```
teager@Tommy ~ % java -version

java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)
```

**Reply by marcprux** [reply, marked as answer]
Ahh, `/usr/local/` indicates that you are using the version of Homebrew for an Intel machine. I expect it is an installation held over from before you upgraded to an M1 Mac. This is a common issue. The easiest solution is to just in-install Homebrew and then re-install it from scratch (which will install the M1 version), and then install `skip` again. After that `skip checkup` should pass, but do let us know if it continues to cause errors.

**Comment by tom-eager** [comment]
I completely uninstalled brew and  I did a new install of  Brew on my M1 Mac. I installed skip successfully and skip checkup passed.  I verified that my license for skip was still valid.  I can now review the skip tools.  Thanks for your help

**Comment by marcprux** [comment]
Glad that helped! Sorry for the trouble … the mismatch with Homebrew installation architectures has been an ongoing headache. We will look into ways for `skip checkup` to give you a better description of the problem.
